### PR TITLE
READY : Implement 3 dimension vector

### PR DESCRIPTION
**Pull request by qarux**
The code needs to be refactored after #39 will be merged (there are a lot of boilerplate trait implementations).

### CODE REVIEW

File: rust/impl/math/adapter/plugin/cgmath/x3.rs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  use crate::prelude::*;
+  use crate::
+  {
+    X3,
+    ScalarInterface
+  };
+
+  //
+
+  impl< Scalar > X3NominalInterface for cgmath::Vector3< Scalar >
```
Comment (dmvict) [inline rust/impl/math/adapter/plugin/cgmath/x3.rs:14]: Why here is many of such code?

File: rust/impl/math/adapter/plugin/cgmath/x3.rs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+      {
+        std::mem::transmute::< _, _ >( self )
+      }
+    }
+  }
+
+  //
+
+  impl< Scalar > X3NominalInterface for cgmath::Point3< Scalar >
+    where
```
Comment (dmvict) [inline rust/impl/math/adapter/plugin/cgmath/x3.rs:114]: Please, fix formatting for such cases.

**Comment by dmvict** [comment]
Please, resolve conflict.

File: rust/impl/math/adapter/plugin/cgmath/x3.rs (hunk 2 above)
Comment (qarux) [reply rust/impl/math/adapter/plugin/cgmath/x3.rs]: Fixed

File: rust/impl/math/adapter/plugin/cgmath/x3.rs (hunk 1 above)
Comment (qarux) [reply rust/impl/math/adapter/plugin/cgmath/x3.rs]: Replaced with macro.
As I mentioned in the description, #39 needed to be merged.

**Comment by dmvict** [comment]
Please, give me the answer. 

Can the procedural macro from #42 be used instead of the declarative macro? 

If not, you should extend the procedural macro.

**Pull request merged**